spec/controllers/cargos_controller_spec.rb

```diff
@@ -28,64 +27,115 @@
         expect(assigns(:cargos)).to eq(Cargo.all.order(price: :desc))
       end
 
-      it 'returns status' do
+      it 'returns status 200' do
         expect(index).to have_http_status(200)
+      end
+
+      context 'when paginate' do
+        subject(:index_paginate) {get :index, params: { page: '2' }}
+        let!(:cargo_list) { create_list :cargo, 5, user_id: user.id }
+        it 'return paginated cargos' do
+          index_paginate
+          expect(assigns(:cargos).count).to eq(2)
+        end
+      end
     end
-  end
 
     context 'when user unauthorized' do
+      before { sign_out user }
+
+      it 'return nothing' do
+        index
+        expect(assigns(:cargos)).to eq(nil)
+      end
+
+      it 'redirects to the login page' do
+        expect(index).to redirect_to(new_user_session_path)
+      end
 
+      it 'return status 302' do
+        expect(index).to have_http_status(302)
+      end
     end
   end
 
   context '#show' do
+    subject(:show) { get :show, params: { id: Cargo.last.id } }
 
     context 'when user authorized' do
       before { sign_in user }
-      subject(:show) { get :show, params: { id: Cargo.last.id } }
 
       it 'returns cargo' do
         show
         expect(assigns(:cargo)).to eq(Cargo.last)
       end
-      
-      it 'returns status' do
+
+      it 'returns status 200' do
         show
         expect(show).to have_http_status(200)
       end
     end
 
     context 'when user unauthorized' do
+      before { sign_out user }
+
+      it 'return nothing' do
+        show
+        expect(assigns(:cargo)).to eq(nil)
+      end
+
+      it 'redirects to the login page' do
+        expect(show).to redirect_to(new_user_session_path)
+      end
 
+      it 'return status 302' do
+        expect(show).to have_http_status(302)
+      end
     end
   end
 
   context '#create' do
+    let!(:cargo_values) { attributes_for :cargo }
+    subject(:create_cargo) { post :create, params: { cargo: cargo_values } }
 
     context 'when user authorized' do
       before { sign_in user }
-      let!(:cargo_values) { attributes_for :cargo }
-      subject(:create_cargo) { post :create, params: { cargo: cargo_values } }
 
       it 'return created cargo' do
         VCR.use_cassette('Cargo_controller') do
           expect { create_cargo }.to change(Cargo, :count).by(1)
         end
       end
```

Comment:
This 'it' doesn't return cargo

Reply:
you mean, 'it' should it be written differently?

Reply:
Yes, 'increase cargos count' or 'creates cargo' for example